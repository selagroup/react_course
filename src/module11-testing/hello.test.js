import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import pretty from "pretty";
import Hello from "./hello";

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});
describe("Component test",() => {
    it("renders with or without a name", () => {
      act(() => {
        render(<Hello />, container);
      });
      expect(container.textContent).toBe("Hey, stranger");
    
      act(() => {
        render(<Hello name="Jenny" />, container);
      });
      expect(container.textContent).toBe("Hello, Jenny!");
    
      act(() => {
        render(<Hello name="Margaret" />, container);
      });
      expect(container.textContent).toBe("Hello, Margaret!");
    });
});

//describe.only
describe("Component HELLO snapshot test",() => {

  it("should render a greeting", () => {
    act(() => {
      render(<Hello />, container);
    });
  
    expect(
      pretty(container.innerHTML)
    ).toMatchSnapshot(); 
  
    act(() => {
      render(<Hello name="JennyZZZZ" />, container);
    });
  
    expect(
      pretty(container.innerHTML)
    ).toMatchSnapshot(); 
  
    
  });
});