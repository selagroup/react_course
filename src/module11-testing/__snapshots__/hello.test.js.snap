// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component HELLO snapshot test should render a greeting 1`] = `"<span>Hey, stranger</span>"`;

exports[`Component HELLO snapshot test should render a greeting 2`] = `"<h1>Hello, JennyZZZZ!</h1>"`;
